src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java

```diff
@@ -0,0 +1,85 @@
+package com.conveyal.datatools.manager.controllers.api;
+
+import com.conveyal.datatools.DatatoolsTest;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static com.conveyal.datatools.TestUtils.parseJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static com.zenika.snapshotmatcher.SnapshotMatcher.matchesSnapshot;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * These tests verify that various Auth0 API calls behave as expected. The Auth0 server is mocked in order to return
+ * certain responses needed to verify functionality.
+ */
+public class UserControllerTest {
+    /**
+     * This sets up a mock server that accepts requests and sends predefined responses to mock an Auth0 server.
+     */
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(
+        options()
+            .port(8089)
```

Comment:
I don't love that the port is set separately from the `AUTH0_DOMAIN` value in the test env.yml. I also don't love the proliferation of config files, especially when only two values are changed in order to test the user controller. Maybe an alternative to simplify things would be to add a `configureAuth0` method in the UserController. We could call this within the `register` method with the values from the config, but then recall it in the `UserControllerTest#setUp` method (after `DataToolsTest.setUp()` of course) to override the values with `testing-client-id` and `localhost:8089`. That way the testing stuff doesn't bleed into the controller as mentioned in my previous comment.

Reply:
Just pushed some changes that hopefully address this comment.